share/vss/pedersen/vss.go

```diff
@@ -408,7 +405,8 @@ func (v *Verifier) decryptDeal(e *EncryptedDeal) (*Deal, error) {
 	if err != nil {
 		return nil, err
 	}
-	decrypted, err := gcm.Open(nil, e.Nonce, e.Cipher, v.hkdfContext)
+	nonce := make([]byte, gcm.NonceSize())
```

Comment:
is it intentional to pass in an empty slice here? it seems like it'd be better to remove it entirely (though this breaks the interface for other users potentially)

Reply:
I'm not sure I understood well your point but I'll try to answer anyway. 

It's not possible to not pass the nonce, as it's required from the gcm interfact; additionally, if we were to pass nil instead that wouldn't work either, as it has to match the NonceSize(). Also, this is a v4 change, so it's okay to break our interfaces (the encrypted deal in this case).